php_phongo.c

```diff
@@ -590,48 +590,79 @@ int phongo_execute_query(mongoc_client_t *client, const char *namespace, zval *z
 	return true;
 } /* }}} */
 
+static bson_t *create_wrapped_command_envelope(bson_t *reply)
+{
+	bson_t *tmp;
+
+	tmp = BCON_NEW("cursor", "{", "id", BCON_INT64(0), "ns", BCON_UTF8("db.$cmd"), "firstBatch", "[", BCON_DOCUMENT(reply), "]", "}");
+
+	return tmp;
+}
+
 int phongo_execute_command(mongoc_client_t *client, const char *db, zval *zcommand, zval *zreadPreference, int server_id, zval *return_value, int return_value_used TSRMLS_DC) /* {{{ */
 {
 	const php_phongo_command_t *command;
-	mongoc_cursor_t *cursor;
 	bson_iter_t iter;
+	bson_t reply;
+	bson_error_t error;
+	bson_t *opts;
+	mongoc_cursor_t *cmd_cursor;
+	uint32_t selected_server_id;
 
 	command = Z_COMMAND_OBJ_P(zcommand);
 
-	cursor = mongoc_client_command(client, db, MONGOC_QUERY_NONE, 0, 1, 0, command->bson, NULL, phongo_read_preference_from_zval(zreadPreference TSRMLS_CC));
+	opts = bson_new();
+	if (server_id > 0) {
```

Comment:
We discussed moving server selection into `Manager.c` so that this function always takes a positive `server_id`. When that time comes, I expect this will simply append `serverId` to `opts` and we'll remove the conditional.

Just mentioning it for the record. There's no reason to make that change in this PR.

Reply:
Ack